### DISCUSSION TITLE
[v5] How to handle conditional relationship based on current row column value

### DISCUSSION THREAD

**Discussion by shunnmugam** [adonisjs/core#2862] [Help] [answered]
How to add relationship based on creent row column value, for example if column data_type is 1 then call question relationship or call video relationship.

bellow one is not working

```b.preload('dataMap', (b) => {
b.where((b) => {
     b.where('data_type', 1)
            .preload('question');
    }).orWhere((b) => {
        b.where('data_type', 2)
            .preload('video');
    })
    .orderBy('order');
})

**Comment by arthurcvm** [comment]
Hi @shunnmugam, whats the return of this?

**Reply by shunnmugam** [reply]
@arthurcvm it's not fetching question and video

**Reply by feniljariwala82** [reply]
Any error or just a blank response?

**Reply by shunmugam-solid** [reply]
blank response

**Comment by RomainLanz** [comment, marked as answer]
Hey @shunnmugam! 👋 

You better of running two queries for your use case.
It will be simpler and easier for you to get back to it later on than writing a complex query.
